src/internal.c

```diff
@@ -11160,7 +11160,11 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                 #ifdef HAVE_OCSP
                     if (doLookup && ssl->ctx->cm->ocspEnabled) {
                         WOLFSSL_MSG("Doing Leaf OCSP check");
-                        ret = CheckCertOCSP_ex(ssl->ctx->cm->ocsp,
+                        /* trust for the lifetime of the responder's cert*/
```

Comment:
This section of the RFC 2560 only applies to CA's. Where are you checking this is a CA?  

Your logic is fully skipping the OCSP check, which is not correct. The fix is to make the use of this extension non-critical and allow NULL. Reviewing this issue it appears having this extension OID set will allow skipping the CRL check.